### PR TITLE
Remove some Windows specific formatters

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
This change has two primary sets of changes, remove the `%I64`/`%I` and `%S` from format strings that exist in unmanaged code. All of the `%I*` formats have been removed, but many `%S` still remain (approx 150). The latter must be fully removed before we can delete our private copy of `printf`.

/cc @janvorli @elinor-fung @am11 @jkotas

### CODE REVIEW

**Review by jkotas — APPROVED** [review]

File: src/coreclr/vm/gcheaputilities.cpp [hunk 1]
```diff
@@ -169,7 +169,7 @@ HMODULE LoadStandaloneGc(LPCWSTR libFileName)
     libPath.Append(libFileName);
 
     LPCWSTR libraryName = libPath.GetUnicode();
-    LOG((LF_GC, LL_INFO100, "Loading standalone GC from path %S\n", libraryName));
```
Comment (davidwrighton) [inline src/coreclr/vm/gcheaputilities.cpp:172]: This looks sketchy. If you call GetUnicode, and then GetUTF8(), I would expect the GetUnicode() pointer to become invalid.

Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/gcheaputilities.cpp:172]: Yes, that is a good catch. I need to audit these changes again.

File: src/coreclr/pal/src/cruntime/printfcpp.cpp [hunk 2]
```diff
@@ -359,7 +359,7 @@ BOOL Internal_ExtractFormatA(CPalThread *pthrCurrent, LPCSTR *Fmt, LPSTR Out, LP
                 *Out++ = '1';
                 *Out++ = '6';
             }
-            /* native *printf does not support %I64p
+            /* native *printf does not support %p
```
Comment (janvorli) [inline src/coreclr/pal/src/cruntime/printfcpp.cpp:362]: A nit - this comment is not true now, the native printf does support %p.

File: src/coreclr/tools/StressLogAnalyzer/StressLogPlugin.cpp [hunk 3]
```diff
@@ -1397,11 +1400,11 @@ int ProcessStressLog(void* baseAddress, int argc, char* argv[])
 
     CorClrData corClrData(hdr);
     FILE* outputFile = stdout;
-    if (s_outputFileName != nullptr)
+    if (s_outputFileNameW != nullptr)
     {
-        if (_wfopen_s(&outputFile, s_outputFileName, W("w")) != 0)
+        if (_wfopen_s(&outputFile, s_outputFileNameW, W("w")) != 0)
```
Comment (janvorli) [inline src/coreclr/tools/StressLogAnalyzer/StressLogPlugin.cpp:1405]: Can't we just use fopen_s now that we have the 8 bit char version of the filename?

**Review by janvorli — COMMENTED** [review]
I just gave it a quick look, I'll take another one tomorrow.

File: src/coreclr/tools/StressLogAnalyzer/StressLogPlugin.cpp (hunk 3 above)
Comment (AaronRobinsonMSFT) [reply src/coreclr/tools/StressLogAnalyzer/StressLogPlugin.cpp:1405]: Good question. Let me see. I'd prefer that anyways.

File: src/coreclr/vm/gcheaputilities.cpp (hunk 1 above)
Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/gcheaputilities.cpp:172]: I went through the rest of these changes. None seem to suffer from the same behavior.

File: src/coreclr/tools/StressLogAnalyzer/StressLogPlugin.cpp [hunk 4]
```diff
@@ -779,10 +779,7 @@ bool ParseOptions(int argc, char* argv[])
             case 'O':
                 if (arg[2] == ':')
                 {
-                    WCHAR* buffer = new WCHAR[1000];
-                    if (MultiByteToWideChar(CP_ACP, 0, &arg[3], -1, buffer, 1000) == 0)
```
Comment (am11) [inline src/coreclr/tools/StressLogAnalyzer/StressLogPlugin.cpp:783]: Just noticed a similar method `convertC` which is unused in `src/mono/dlls/dbgshim/dbgshim.cpp` (can be deleted).

File: src/coreclr/vm/multicorejit.cpp [hunk 5]
```diff
@@ -919,7 +919,7 @@ HRESULT MulticoreJitRecorder::StopProfile(bool appDomainShutdown)
         hr = WriteOutput();
     }
 
-    MulticoreJitTrace(("StopProfile: Save new profile to %S, hr=0x%x", m_fullFileName.GetUnicode(), hr));
+    MulticoreJitTrace(("StopProfile: Save new profile to %s, hr=0x%x", m_fullFileName.GetUTF8(), hr));
```
[the remark this answers is not in the document]
Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/multicorejit.cpp:922]: Good point. I reverted this change due to other issues.

File: src/coreclr/vm/multicorejit.cpp [hunk 6]
```diff
@@ -1049,7 +1049,7 @@ HRESULT MulticoreJitRecorder::StartProfile(const WCHAR * pRoot, const WCHAR * pF
                 player.SuppressRelease();
             }
 
-            MulticoreJitTrace(("ProcessProfile('%S') returns %x", m_fullFileName.GetUnicode(), hr1));
+            MulticoreJitTrace(("ProcessProfile('%s') returns %x", m_fullFileName.GetUTF8(), hr1));
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/vm/multicorejit.cpp:1052]: Reverted.

File: src/coreclr/vm/multicorejit.cpp [hunk 7]
```diff
@@ -1049,7 +1049,7 @@ HRESULT MulticoreJitRecorder::StartProfile(const WCHAR * pRoot, const WCHAR * pF
                 player.SuppressRelease();
             }
 
-            MulticoreJitTrace(("ProcessProfile('%S') returns %x", m_fullFileName.GetUnicode(), hr1));
+            MulticoreJitTrace(("ProcessProfile('%s') returns %x", m_fullFileName.GetUnicode(), hr1));
```
Comment (davidwrighton) [inline src/coreclr/vm/multicorejit.cpp:1052]: Why is it ok to use %s here instead of %S?

Comment (AaronRobinsonMSFT) [inline src/coreclr/vm/multicorejit.cpp:1052]:
```suggestion
            MulticoreJitTrace(("ProcessProfile('%S') returns %x", m_fullFileName.GetUnicode(), hr1));
```

**Comment by AaronRobinsonMSFT** [comment]
@janvorli I think I've addressed your feedback. Any other concerns?

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Pull request merged**